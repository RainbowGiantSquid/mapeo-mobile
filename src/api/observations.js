// @flow
import { jsonRequest } from './base';
import { keyBy } from 'lodash';
import { parseObservationResponse } from '../models/observations';
import querystring from 'querystring'

import type {
  Observation as ObservationType,
  ServerObservationResponse,
  ServerObservationCreate
} from '../types/observation';

class Observation {
  static list = (opts) =>
    jsonRequest({
      method: 'GET',
      route: '/observations?' + querystring.stringify(opts)
    }).map(observations => {
      var payload = observations.map(parseObservationResponse)
      return keyBy(payload, 'id')
    });

  static create = (observation: ServerObservationCreate) =>
    jsonRequest({
      method: 'POST',
      route: '/observations',
      body: observation
    }).map(parseObservationResponse);

  static update = (observation: ServerObservationResponse) =>
    jsonRequest({
      method: 'PUT',
      route: `/observations/${observation.id}`,
      body: observation
    }).map(parseObservationResponse);
}

export default Observation;
