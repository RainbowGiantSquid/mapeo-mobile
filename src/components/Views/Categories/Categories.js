// @flow
import React from 'react';
import MapboxGL from '@mapbox/react-native-mapbox-gl';
import {
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
  FlatList,
  Dimensions,
  Image
} from 'react-native';
import { NavigationActions, withNavigation } from 'react-navigation';
import Icon from 'react-native-vector-icons/MaterialIcons';
import type { Category } from '../../../types/category';
import type { Observation } from '../../../types/observation';
import { DARK_GREY, LIGHT_GREY, WHITE, CHARCOAL } from '../../../lib/styles';

type Props = {
  navigation: NavigationActions
};

export type StateProps = {
  categories: Category[],
  selectedObservation: Observation
};

export type DispatchProps = {
  listCategories: () => void,
  updateObservation: (obs: Observation) => void,
  resetNavigation: () => void
};

const styles = StyleSheet.create({
  header: {
    flexDirection: 'row',
    flex: 1,
    backgroundColor: DARK_GREY,
    paddingVertical: 20,
    alignItems: 'center',
    justifyContent: 'center'
  },
  close: {
    position: 'absolute',
    left: 10
  },
  title: {
    fontSize: 20,
    fontWeight: '700',
    color: LIGHT_GREY
  },
  circle: {
    width: 50,
    height: 50,
    backgroundColor: 'white',
    borderRadius: 50,
    borderColor: LIGHT_GREY,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: 'red',
    shadowRadius: 10,
    shadowOpacity: 1,
    marginBottom: 5
  },
  innerCircle: {
    width: 40,
    height: 40,
    backgroundColor: LIGHT_GREY,
    borderRadius: 50,
    alignItems: 'center',
    justifyContent: 'center'
  },
  cellContainer: {
    flex: 1,
    height: 130,
    paddingTop: 20,
    paddingBottom: 20,
    width: Dimensions.get('window').width / 3
  },
  cell: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center'
  },
  categoryName: {
    color: 'black',
    fontWeight: '700',
    textAlign: 'center',
    maxWidth: Dimensions.get('window').width / 3
  }
});

class Categories extends React.PureComponent<
  StateProps & Props & DispatchProps
> {
  componentDidMount() {
    const { listCategories } = this.props;

    listCategories();
  }

  map: any;
  handleUpdateObservation = item => {
    const { updateObservation, selectedObservation } = this.props;

    updateObservation({
      ...selectedObservation,
      icon: item.icon
    });
    this.props.navigation.navigate('ObservationEditor', {
      category: item.id
    });
  };

  renderHeader = () => {
    const { navigation } = this.props;

    return (
      <TouchableHighlight
        style={{ flexDirection: 'row' }}
        onPress={() => navigation.navigate('Position')}
      >
        <View style={styles.header}>
          <Icon style={styles.close} color="gray" name="close" size={25} />
          <Text style={styles.title}>Categor√≠a</Text>
        </View>
      </TouchableHighlight>
    );
  };

  renderItem = ({ item }) => {
    const { resetNavigation } = this.props;

    return (
      <TouchableHighlight
        style={styles.cellContainer}
        underlayColor="transparent"
        onPress={() => {
          resetNavigation();
          this.handleUpdateObservation(item);
        }}
      >
        <View style={styles.cell}>
          <View style={styles.circle}>
            <View style={styles.innerCircle}>{item.icon}</View>
          </View>
          <Text style={styles.categoryName}>{item.name}</Text>
        </View>
      </TouchableHighlight>
    );
  };

  render() {
    const { categories, selectedObservation } = this.props;
    const keyExtractor = item => item.id;

    return (
      <View style={{ flex: 1 }}>
        <MapboxGL.MapView
          style={{ flex: 1 }}
          centerCoordinate={
            selectedObservation
              ? [selectedObservation.lon, selectedObservation.lat]
              : undefined
          }
          ref={c => (this.map = c)}
          zoomLevel={12}
          logoEnabled={false}
        >
          {selectedObservation && (
            <MapboxGL.PointAnnotation
              id="selected"
              coordinate={[selectedObservation.lon, selectedObservation.lat]}
              selected
            >
              <MapboxGL.Callout>
                <View
                  style={{
                    backgroundColor: WHITE,
                    alignItems: 'center',
                    justifyContent: 'center',
                    padding: 10,
                    borderRadius: 5
                  }}
                >
                  {!!selectedObservation.media.length && (
                    <Image
                      style={{ width: 50, height: 50, marginBottom: 10 }}
                      source={{ uri: selectedObservation.media[0].source }}
                    />
                  )}
                  <Text style={{ color: CHARCOAL }}>
                    {selectedObservation.lon}, {selectedObservation.lat}
                  </Text>
                </View>
              </MapboxGL.Callout>
            </MapboxGL.PointAnnotation>
          )}
        </MapboxGL.MapView>
        <FlatList
          style={{
            height: 80,
            width: Dimensions.get('window').width,
            backgroundColor: WHITE
          }}
          keyExtractor={keyExtractor}
          renderItem={this.renderItem}
          data={categories}
          numColumns={3}
        />
      </View>
    );
  }
}

export default withNavigation(Categories);
