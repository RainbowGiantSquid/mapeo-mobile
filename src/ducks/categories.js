// @flow
import { create } from '@src/lib/redux';
import update from 'immutability-helper';
import { keyBy } from 'lodash';

export const {
  type: CATEGORY_LIST,
  action: categoryList,
  reducer: categoryListReducer
} = create('CATEGORY_LIST', {
  success: (state, action) => {
    const newState = update(state, {
      categories: { $set: keyBy(action.payload, 'id') }
    });

    return newState;
  }
});

export default [categoryListReducer];
