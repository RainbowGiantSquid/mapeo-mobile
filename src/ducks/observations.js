// @flow
import { create } from '@src/lib/redux';
import update from 'immutability-helper';
import { keyBy } from 'lodash';
import { createSelector } from '@src/lib/selector';
import { StoreState } from '@types/redux';
import { Observation } from '@types/observation';

export const {
  type: OBSERVATION_LIST,
  action: observationList,
  reducer: observationListReducer,
} = create('OBSERVATION_LIST', {
  success: (state, action) => {
    const newState = update(state, {
      observations: { $set: keyBy(action.payload, 'id') },
    });

    return newState;
  },
});

export const selectContact = createSelector(
  [
    (state: StoreState, id: string): Observation => state.app.observations[id],
  ],
  (observation: Observation): Observation => observation,
  (observation: Observation, id: string): string => id,
);

export default [observationListReducer];
