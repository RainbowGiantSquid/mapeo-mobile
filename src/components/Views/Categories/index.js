// @flow
import { connect } from 'react-redux';
import { StoreState } from '@types/redux';
import type { Dispatch } from 'redux';
import { values } from 'lodash';
import { categoryList } from '@ducks/categories';
import Categories from './Categories';
import type { StateProps } from './Categories';

function mapStateToProps(state: StoreState): StateProps {
  const categories = values(state.app.categories).sort(
    (a, b) => a.name - b.name
  );

  return { categories };
}

function mapDispatchToProps(dispatch: Dispatch) {
  return { listCategories: () => dispatch(categoryList('')) };
}

export default connect(mapStateToProps, mapDispatchToProps)(Categories);
