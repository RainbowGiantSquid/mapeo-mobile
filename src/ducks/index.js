import { combineReducers } from 'redux';
import { combineReducers as appCombineReducers } from '@src/lib/redux';
import MainStackNavigation from '@src/components/MainNavigation/MainStackNavigation';
import TabBarNavigation from '@src/components/Views/TabBarView';

import categories from './categories';
import observations from './observations';

const rootReducer = combineReducers({
  app: appCombineReducers(...categories, ...observations),

  mainStack: (state, action) =>
    MainStackNavigation.router.getStateForAction(action, state),

  tabBar: (state, action) =>
    TabBarNavigation.router.getStateForAction(action, state)
});

export default rootReducer;
