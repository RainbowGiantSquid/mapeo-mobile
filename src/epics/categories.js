// @flow
import React from 'react';
import type { ActionsObservable } from 'redux-observable';
import { Image } from 'react-native';
import { CATEGORY_LIST, categoryList } from '../ducks/categories';
import { createCategory } from '../mocks/categories';
import type { Action } from '../types/redux';
import type { Category } from '../types/category';

const initialCategories = [
  {
    name: 'Rios y corrientes',
    icon: (
      <Image
        source={require('../images/categories/category_0.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Túneles / cuevas',
    icon: (
      <Image
        source={require('../images/categories/category_1.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Sitios de caza',
    icon: (
      <Image
        source={require('../images/categories/category_2.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Sitios de pesca',
    icon: (
      <Image
        source={require('../images/categories/category_3.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Casas antiguas y granjas',
    icon: (
      <Image
        source={require('../images/categories/category_4.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Cementerios',
    icon: (
      <Image
        source={require('../images/categories/category_5.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Plantas medicinales',
    icon: (
      <Image
        source={require('../images/categories/category_6.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Recursos para artesania',
    icon: (
      <Image
        source={require('../images/categories/category_7.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Oro',
    icon: (
      <Image
        source={require('../images/categories/category_8.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Árboles frutales plantados',
    icon: (
      <Image
        source={require('../images/categories/category_9.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Yoko',
    icon: (
      <Image
        source={require('../images/categories/category_10.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Yage',
    icon: (
      <Image
        source={require('../images/categories/category_11.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Gente invisible',
    icon: (
      <Image
        source={require('../images/categories/category_12.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Cascadas',
    icon: (
      <Image
        source={require('../images/categories/category_13.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Colinas y montañas (sagrado)',
    icon: (
      <Image
        source={require('../images/categories/category_14.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Enormes árboles especiales',
    icon: (
      <Image
        source={require('../images/categories/category_15.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Boas',
    icon: (
      <Image
        source={require('../images/categories/category_16.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Lugares históricos',
    icon: (
      <Image
        source={require('../images/categories/category_17.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  },
  {
    name: 'Plantas y animales en peligro de extinció',
    icon: (
      <Image
        source={require('../images/categories/category_18.png')}
        style={{ width: 40, height: 40 }}
        resizeMode="contain"
      />
    )
  }
];

export const categoryListEpic = (
  action$: ActionsObservable<Action<string, Category[]>>
) =>
  action$
    .ofType(CATEGORY_LIST)
    .filter(action => action.status === 'Start')
    .map(() =>
      categoryList(
        '',
        initialCategories.map((c, i) =>
          createCategory({
            name: c.name,
            id: i,
            icon: c.icon
          })
        )
      )
    );

export default [categoryListEpic];
