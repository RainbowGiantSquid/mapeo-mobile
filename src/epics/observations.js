// @flow
import React from 'react';
import { Image } from 'react-native';
import type { ActionsObservable } from 'redux-observable';
import { OBSERVATION_LIST, observationList } from '../ducks/observations';
import { Action } from '../types/redux';
import { Observation } from '../types/observation';

export const observationListEpic = (
  action$: ActionsObservable<Action<string, Observation[]>>,
  store: any
) =>
  action$
    .ofType(OBSERVATION_LIST)
    .filter(
      action =>
        action.status === 'Start' &&
        store.getState().app &&
        store.getState().app.observations &&
        !store.getState().app.observations.length
    )
    .map(() =>
      observationList('', [
        {
          type: 'Old houses and farms',
          id: 1,
          lat: 0.106831547,
          lon: -77.42499473,
          link: 'link',
          created: new Date('2018-03-25T12:00:00Z'),
          name: 'Old houses and farms',
          notes: `Pedro and Maria's house`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/1-house.jpg')
            }
          ],
          icon: require('../images/categories/category_0.png'),
          mock: true
        },
        {
          type: 'Hunting sites',
          id: 2,
          lat: 0.088170932,
          lon: -77.44020885,
          link: 'link',
          created: new Date('2018-03-25T12:30:00Z'),
          name: 'Hunting sites',
          notes: `Don Pancho's hunting camp`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/4-camp.jpg')
            }
          ],
          icon: require('../images/categories/category_2.png'),
          mock: true
        },
        {
          type: 'Threat',
          id: 3,
          lat: 0.112411067,
          lon: -77.41378756,
          link: 'link',
          created: new Date('2018-02-03T17:30:00Z'),
          name: 'Threat',
          notes: `Oil spill 3 February 2018. Ruptured pipeline. Dead fish, oily water`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/5-threat.jpg')
            }
          ],
          icon: null,
          mock: true
        },
        {
          type: 'Medicinal plants',
          id: 4,
          lat: 0.09749025,
          lon: -77.43614404,
          link: 'link',
          created: new Date('2018-02-04T17:30:00Z'),
          name: 'Medicinal plants',
          notes: `Salola fruit plant - used in tea to heal inflamation`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/6-plant.jpg')
            }
          ],
          icon: require('../images/categories/category_6.png'),
          mock: true
        },
        {
          type: 'Waterfalls',
          id: 5,
          lat: 0.105818148,
          lon: -77.4408037,
          link: 'link',
          created: new Date('2018-03-04T17:30:00Z'),
          name: 'Waterfalls',
          notes: `Sacred waterfall, visited during vision quests for ancestors, for taking ayahuasca`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/7-waterfall.jpg')
            }
          ],
          icon: require('../images/categories/category_13.png'),
          mock: true
        },
        {
          type: 'Rivers and streams',
          id: 6,
          lat: 0.119004333,
          lon: -77.42374225,
          link: 'link',
          created: new Date('2018-03-24T17:30:00Z'),
          name: 'Rivers and streams',
          notes: `Point where pipeline crosses river`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/8-threat.jpg')
            }
          ],
          icon: require('../images/categories/category_0.png'),
          mock: true
        },
        {
          type: 'Historical sites',
          id: 7,
          lat: 0.082856811,
          lon: -77.4429836,
          link: 'link',
          created: new Date('2018-03-24T12:30:00Z'),
          name: 'Historical sites',
          notes: `There used to be a village here in the 1940s, it was abandoned after an epidemic of flu which killed most of the elders and children.`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/9-specialsite.jpg')
            }
          ],
          icon: require('../images/categories/category_17.png'),
          mock: true
        },
        {
          type: 'Gold',
          id: 8,
          lat: 0.108605553,
          lon: -77.41560494,
          link: 'link',
          created: new Date('2018-03-11T17:30:00Z'),
          name: 'Gold',
          notes: `Illegal gold miners 11 March 2018, excavators, dredges, open tailing pond, mercury.`,
          observedBy: 'Aliya',
          media: [
            {
              type: 'LocalPhoto',
              source: require('../images/mock-data/10-threat.jpg')
            }
          ],
          icon: require('../images/categories/category_8.png'),
          mock: true
        }
      ])
    );

export default [observationListEpic];
