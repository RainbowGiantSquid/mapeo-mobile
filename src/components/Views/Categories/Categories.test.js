// @flow
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import Categories from './Categories';
import { categoryList } from '../../../ducks/categories';
import { createObservation } from '../../../mocks/observations';
import { createCategory } from '../../../mocks/categories';

describe('Categories tests', () => {
  const isFocused = () => true;
  const addListener = () => true;
  const categories = [createCategory()];
  const selectedObservation = createObservation();
  jest.mock(`react-native-camera`, () => {
    const React = require('React');
  });

  test('snapshots', () => {
    let tree;
    tree = shallow(
      <Categories
        categories={categories}
        selectedObservation={selectedObservation}
        navigation={{ isFocused, addListener }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
