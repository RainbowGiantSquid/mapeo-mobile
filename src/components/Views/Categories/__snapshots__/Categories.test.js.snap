// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories tests snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <Header
        leftIcon={
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            underlayColor="rgba(0, 0, 0, 0.5)"
          >
            <Icon
              allowFontScaling={false}
              color="#9E9C9C"
              name="window-close"
              size={30}
            />
          </TouchableOpacity>
        }
        rightIcon={
          <Component
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#d6d2cf",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "width": 30,
              }
            }
          >
            <Component
              style={
                Object {
                  "backgroundColor": "#EAEAEA",
                  "borderRadius": 50,
                  "height": 25,
                  "justifyContent": "center",
                  "width": 25,
                }
              }
            >
              <Icon
                allowFontScaling={false}
                color="white"
                name="check"
                size={18}
                style={
                  Object {
                    "alignSelf": "center",
                    "marginLeft": 3,
                  }
                }
              />
            </Component>
          </Component>
        }
      />
    }
    data={
      Array [
        Object {
          "id": "categoryID",
          "name": "categoryName",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    invertStickyHeaders={undefined}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={3}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#ccffff",
        "height": 80,
        "width": 750,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "paddingHorizontal": 15,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#9E9C9C",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(0, 0, 0, .8)",
                  "borderRadius": 50,
                  "flexDirection": "row",
                  "height": 35,
                  "paddingLeft": 13,
                  "paddingRight": 15,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#7AFA4C",
                    "borderRadius": 50,
                    "height": 10,
                    "width": 10,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "marginHorizontal": 20,
                  }
                }
              >
                +/- 100m
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#d6d2cf",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "width": 30,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#EAEAEA",
                  "borderRadius": 50,
                  "height": 25,
                  "justifyContent": "center",
                  "width": 25,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 18,
                    },
                    Object {
                      "alignSelf": "center",
                      "marginLeft": 3,
                    },
                    Object {
                      "fontFamily": "Octicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "height": 130,
                "opacity": 1,
                "paddingBottom": 20,
                "paddingTop": 20,
                "width": 250,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "black",
                    "fontWeight": "700",
                    "maxWidth": 248,
                    "textAlign": "center",
                  }
                }
              >
                [missing "en.categories.categoryID" translation]
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
